addon/model.js

```diff
@@ -331,12 +433,22 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   get id() {
-    return this._internalModel.id;
+    if (false) {
+      if (!this._recordData) {
+        return null;
+      }
+      return this._recordData.id;
+    } else {
+      return this._internalModel.id;
+    }
   }
 
   set id(value) {
+    //TODO need a test for this
```

Comment:
let's add a test then

Reply:
I don't think in custom model classes this code path is relevant anymore, will try to see if there is a high level test that would work as a follow up

Reply:
>  will try to see if there is a high level test that would work as a follow up

what was the result of doing so?